Automattic/delphin
# Contact Information: add form validation

This pull request adds validation to the form on the `Contact Information` page. The actual validation is performed by the WordPress.com API.
#### Testing instructions
1. Run `git checkout add/contact-information-validation` and start your server
2. Log into your account
3. Open the [`Search` page](http://delphin.localhost:1337/search) and type some keywords
4. Click a domain suggestion
5. Check that each input field on the `Contact Information` page is validated upon form submission. For instance, we'll display red error messages for required fields (all but `organization` and `address2`) if they are not filled in. Some things that do not need to be resolved here:
- When validation is successful (which never happens right now, because #164 is unresolved), nothing happens. We'll update this to direct users to checkout when we work on checkout.
- The red error messages do not match the styles in #162, and break the layout in the case of city/state/zip. This will be updated when we address that issue.
#### Additional notes

The `/me/domain-contact-information/validate` API endpoint returns errors in two distinct formats:

``` json
{
  "success": false,
  "messages": {
    "first_name": ["The \u201cFirst Name\u201d field is required."],
    "last_name": ["The \u201cLast Name\u201d field is required."],
    "address_1": ["The \u201cAddress 1\u201d field is required."],
    "postal_code": ["The \u201cPostal Code\u201d field is required."],
    "city": ["The \u201cCity\u201d field is required."],
    "country_code": ["The \u201cCountry\u201d field is required."],
    "email": ["The \u201cEmail\u201d field is required."],
    "phone": ["The \u201cPhone\u201d field is required."]
  }
}
```

``` json
{
  "error": "missing_domain_names",
  "message": "Missing domain names parameter"
}
```
#### Reviews
